### PR TITLE
ReferenceLineProvider: support muti navigation paths change lane

### PR DESCRIPTION
**Pull request by mfyanga**
add how to set change lane condition in ReferenceLineProvider::GetReferenceLinesFromRelativeMap
 
 add data for test case in testdata/navigation_mode_test/1_relative_map.pb.txt

### CODE REVIEW

**Comment by mfyanga** [comment]
add how to set change lane condition in ReferenceLineProvider::GetReferenceLinesFromRelativeMap

add data for test case in testdata/navigation_mode_test/1_relative_map.pb.txt

File: modules/planning/reference_line/reference_line_provider.cc [hunk 1]
```diff
@@ -307,7 +440,86 @@ bool ReferenceLineProvider::GetReferenceLinesFromRelativeMap(
...
+  std::copy_if(lanes.begin(), lanes.end(), std::back_inserter(valid_lanes),
+               [&](hdmap::LaneInfoConstPtr ptr) {
+                 return navigation_lane_ids.count(ptr->lane().id().id()) > 0;
+               });
+
+  // get nearest lane wayponints for current adc position
+  double min_distance = std::numeric_limits<double>::infinity();
+  for (const auto &lane : valid_lanes) {
+    // project adc point to lane to check if it is out of lane range
+    {
```
Comment (lianglia-apollo) [inline modules/planning/reference_line/reference_line_provider.cc]: this pair of curly braces are redundant

File: modules/planning/reference_line/reference_line_provider.cc [hunk 2]
```diff
@@ -283,7 +285,129 @@ bool ReferenceLineProvider::GetReferenceLinesFromRelativeMap(
...
+  hdmap::LaneWaypoint adc_lane_way_point;
+  if (!GetNearestWayPointFromNavigationPath(vehicle_state,
+                                            navigation_lane_ids,
+                                            &adc_lane_way_point)) {
+    return false;
+  }
+  const std::string adc_lane_id = adc_lane_way_point.lane->id().id();
+  auto adc_navigation_path = relative_map.navigation_path().find(adc_lane_id);
+  if (adc_navigation_path == relative_map.navigation_path().end()) {
+    AERROR << "adc lane is not find in relative_map.navigation_path";
```
Comment (YajiaZhang) [inline modules/planning/reference_line/reference_line_provider.cc]: "cannot be found"

File: modules/planning/reference_line/reference_line_provider.cc [hunk 3]
```diff
@@ -283,7 +285,129 @@ bool ReferenceLineProvider::GetReferenceLinesFromRelativeMap(
...
+  }
+  // get adc right neighbor lanes
+  for (const auto& right_lane_id :
+       adc_lane_way_point.lane->lane().right_neighbor_forward_lane_id()) {
+     right_neighbor_lane_ids.push_back(right_lane_id.id());
+  }
+
+  // 2.get the higher priority lane info list which priority higher
+  // than current lane and get the highest one as the target lane
+  using Lane_Pair_Type = std::pair<std::string, uint32_t>;
```
Comment (YajiaZhang) [inline modules/planning/reference_line/reference_line_provider.cc]: Google style suggests no underscores in type names.

File: modules/planning/reference_line/reference_line_provider.cc [hunk 4]
```diff
@@ -283,7 +285,129 @@ bool ReferenceLineProvider::GetReferenceLinesFromRelativeMap(
...
+  std::list<Lane_Pair_Type> high_priority_lane_list;
```
Comment (YajiaZhang) [inline modules/planning/reference_line/reference_line_provider.cc]: Use std::vector instead

File: modules/planning/reference_line/reference_line_provider.cc [hunk 5]
```diff
@@ -283,7 +285,129 @@ bool ReferenceLineProvider::GetReferenceLinesFromRelativeMap(
...
+  ADEBUG << "relative_map.navigation_path_size = "
+         << relative_map.navigation_path_size();
+  for (const auto &path_pair : relative_map.navigation_path()) {
+    const auto lane_id = path_pair.first;
+    const uint32_t priority = path_pair.second.path_priority();
+    ADEBUG << "lane_id = " << lane_id;
+    ADEBUG << "priority = " << priority;
+    // the smaller the number, the higher the priority
+    if (adc_lane_id != lane_id && priority < adc_lane_priority) {
+      high_priority_lane_list.emplace_back(std::make_pair(lane_id, priority));
```
Comment (YajiaZhang) [inline modules/planning/reference_line/reference_line_provider.cc]: emplace_back(lane_id, priority)

File: modules/planning/reference_line/reference_line_provider.cc [hunk 6]
```diff
@@ -283,7 +285,129 @@ bool ReferenceLineProvider::GetReferenceLinesFromRelativeMap(
...
+    }
+  }
+  // get the target lane
+  bool is_need_change_lane = false;
```
Comment (YajiaZhang) [inline modules/planning/reference_line/reference_line_provider.cc]: need_lane_change or is_lane_change_needed

File: modules/planning/reference_line/reference_line_provider.cc (hunk 1 above)
Comment (mfyanga) [reply modules/planning/reference_line/reference_line_provider.cc]: ok, thank you

File: modules/planning/reference_line/reference_line_provider.cc (hunk 2 above)
Comment (mfyanga) [reply modules/planning/reference_line/reference_line_provider.cc]: ok, thank you

**Comment by mfyanga** [comment]
Thank you for the review.

Modifications have been made in accordance with the review comments

**Pull request closed without merging**